documentation/configuration/configuration.md
---
description: Those are the basic and most important settings to make.
---

# Config File

## Shop information

Those information will mostly be used for view or email templates.

```php
'name'     => 'Statamic Butik',
'address1' => '132 Main Street',
'address2' => 'Sea view, Fahrensodde', 
'zip_city' => '24937 Flensburg',
'country'  => 'DE',
'phone'    => '+49 1294 238474934',
'mail'     => 'butik@shop.com',
```

{% hint style="info" %}
 You need to define your country via an alpha-2 iso code.  
[https://en.wikipedia.org/wiki/ISO\_3166-1\_alpha-2\#Officially\_assigned\_code\_elements](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements)
{% endhint %}

The defined country will be set as your default country. As long as a user does not select another country in the checkout process, all shipping prices will be calculated on the basis of the default country.

## Shop overview

```php
'overview_type'    => 'newest',
'overview_limit'   => '6',
```

What do you want butik to show on the overview page; the entry point for your shop?

{% hint style="info" %}
By default, the shop entry point will be `/shop`
{% endhint %}

| Options | Description |
| :--- | :--- |
| **all** | All products ordered by name. The limit will be ignored. |
| **name** | All products ordered by name. The limit will be respected. |
| **newest** | Showing the latest added products. |
| **cheapest** | Showing products with the lowest price first. |

The overview limit does only limit the number of products shown on the shop overview page.

## Mail

```php
'order-confirmations' => 'your_mail@butik.com',
```

Butik will send one confirmation e-mail for every sold product. You can define the address right here.

{% hint style="info" %}
You need to set up your mails with Statamic correctly, before butik can successfully notify you. 
{% endhint %}

See the [Statamic documentation](https://statamic.dev/email) for more information. You can easily send yourself a test mail from the control panel to confirm that you mail is working as expected.

## Currency

```php
'currency_name'      => 'Euro',
'currency_isoCode'   => 'EUR',
'currency_symbol'    => '€',
'currency_delimiter' => ',',
```

Choose the currency of your default country. 

Please keep the currency\_isoCode strict to the ISO 4217 standard. That's just a fancy name for the know currency syntax:  
[https://en.wikipedia.org/wiki/ISO\_4217](https://en.wikipedia.org/wiki/ISO_4217)

## Orders

```php
'orders_deletable'  => env('BUTIK_ORDERS_DELETABLE', 'development'),
```

For security reasons, you can define if orders can be deleted and if they can, from whom.

| options | Description |
| :--- | :--- |
| **never** | Orders can't be deleted at all. |
| **development** | Orders can be deleted in development mode only. |
| **users** | Orders can be deleted by all users, if they got the correct Statamic permissions. |

Check out the [orders documentation](orders.md) for more information.

## Widgets

```php
'widget_orders-limit' => 10,
```

The default is set to 10 entries. If you do use the _Orders_ widget and want to regulate the number of orders shown, please do so. 

## Routing

```php
'shop_route_active'     => true,
'product_route_active'  => true, 
'category_route_active' => true,

'route_shop-prefix'       => 'shop',
'route_product'           => '{product}/{variant?}',
'route_cart'              => 'cart',
'route_category'          => 'category/{category}',     // keep {category}         
'route_checkout-delivery' => 'checkout/delivery',          
'route_checkout-payment'  => 'checkout/payment',           
'route_payment-receipt'   => 'payment/{order}/receipt', // keep {order}
```

We made it easy to configure all standard routes. This could be handy in case you want localized routes.

There may be cases where you don't want to use our default routes to view the shop, the product or categories. Set them to false to deactivate those routes.

## Layouts

```php
'layout_product-index'            => 'layout',
'layout_product-category'         => 'layout',
'layout_product-show'             => 'layout',
'layout_cart'                     => 'layout',
'layout_checkout-delivery'        => 'layout',
'layout_checkout-payment'         => 'layout',
'layout_checkout-receipt'         => 'layout',
'layout_checkout-receipt-invalid' => 'layout',
```

We will default to the standard layout in your `resources/views/` folder. Move and/or change the layout file as you need it and define it here. _Butik_ will take of the rest.

## Template

```php
'template_product-index'            => 'butik::web.shop.index',
'template_product-category'         => 'butik::web.shop.category',
'template_product-show'             => 'butik::web.shop.show',
'template_cart'                     => 'butik::web.cart.index',
'template_checkout-delivery'        => 'butik::web.checkout.delivery',
'template_checkout-payment'         => 'butik::web.checkout.payment',
'template_checkout-receipt'         => 'butik::web.checkout.receipt',
'template_checkout-receipt-invalid' => 'butik::web.checkout.invalid-receipt',
```

We put a lot of effort into the creation of our default templates to get you started as quick as possible.

The files will automatically be stored inside `resources/views/vendor/butik`.   
You can move those files or create your own. 

If you want to link to a file named cart inside your `resources/views/shop/cart.blade.php` folder, the updated path would be `shop.cart` without the prefix `butik::`

 [Mike Martin](https://mike-martin.ca/) created our beautiful templates.

## Shipping

```php
'shipping' => [
    'price' => \Jonassiewertsen\StatamicButik\Shipping\ShippingByPrice::class,
],
```

If you want, you can implement your own shipping methods or disable those, you don't want to use.

{% page-ref page="../extending/shipping.md" %}

## Payment Gateway

```php
'payment_gateway' => Jonassiewertsen\StatamicButik\Http\Controllers\PaymentGateways\MolliePaymentGateway::class,
```

Out of the box, _butik_ does implement [Mollie](www.mollie.com/en) as our payment provider of choice, which is ready to use. 

You can easily swap our implementation with your own.

{% page-ref page="../extending/payment-gateway.md" %}





CHANGELOG.md
# Release Notes

## 3.5.4 (09. Mar 2021)
- Translate shop pages correctly (#201)
- Don't gitignore the compsoer.lock file
- Swap hardcoded text in product show view with translation tag. 

## 3.5.3 (22. Feb 2021)
- Cast all order dates correctly
- Language tweaks (#195 thanks @philipboomy)
- Update a doc block typo

## 3.5.2 (06. Feb 2021)
- Only create a show url if the product route config is set to true (#192)
- Hiding the is_visible functionality. (#188)

## 3.5.1 (25. Jan 2021)
- A hot fix for an issue with the products tag.

## 3.5.0 (25. Jan 2021)
- Shipping information will get saved with the order
- Orders can deleted (#143)
- It's configurable if orders can be deleted and by whom.
- Include shipping taxes in total taxes as calculated by the cart (#183)

See the docs for more information:
https://www.butik.dev/configuration/orders

## 3.4.1 (16. Jan 2021)
- Respect saved listing preferences on the orders overview. (#181)
- Add the missing translation attribute 'items count'.
- Make the categories menu responsive for mobile devices. (#171)
- Bumping the lowest Statamic requirment to version 3.0.38.
- Register filters via the newly added $scopes attribute.

## 3.4.0 (13. Jan 2021)
- A new and better overview for orders has been implemented. Orders can be reordered and searched.
- The orders overview UI does look a native collection overview

### Update
Does the folder `resources/lang/vendor/butik` exist? Please update your lang files and add:

```php
'completed_at' => 'Completed at',
'ordered_at' => 'Ordered at',
'order_completed_filter_label' => 'order:completed',
'order_failed_filter_label' => 'order:failed',
'order_paid_filter_label' => 'order:paid',
```

- Does the folder not exist? Don't worry at all :-)


config/config.php
<?php

return [

    /**
     * More detailed information can be found here:
     * https://butik.dev/installation/configuration.
     *
     *
     * SHOP INFORMATION
     *
     * Put your shop information here. Those information will be usedby templates for the shop and emails.
     */
    'name'     => 'Statamic Butik', // Whats the name of your Shop?
    'address1' => '132 Main Street',
    'address2' => 'Sea view, Fahrensodde', // can be left empty
    'zip_city' => '24937 Flensburg',
    'country'  => 'DE', // Set your countries iso code
    'phone'    => '+49 1294 238474934',
    'mail'     => 'butik@shop.com',

    /**
     * Shop overview.
     *
     * What should be displayed on the start page of butik?
     *
     * all:      All products ordered by name. The limit will be ignored.
     * name:     All products ordered by name. The limit will be respected.
     * newest:   Showing the latest added products
     * cheapest: Showing products with the best price first.
     */
    'overview_type'  => 'newest',
    'overview_limit' => 6,

    /**
     * MAIL.
     *
     * Butik will send one confirmation email for every sold product to this address.
     */
    'order-confirmations' => 'your_mail@butik.com',

    /**
     * CURRENCY.
     *
     * Define your shops currency
     */
    'currency_name'                => 'Euro',
    'currency_isoCode'             => 'EUR', // Make sure to use ISO_4217 https://en.wikipedia.org/wiki/ISO_4217
    'currency_symbol'              => '€',
    'currency_delimiter'           => ',',
    'currency_thousands_separator' => '.',

    /**
     * Oders settings.
     *
     * You can define how orders can be deleted. Can they at all, only in production or via the UI
     * if the user has gotten permission to do so.
     *
     * never: Safety first. Orders can never be delete.
     * development: As long as not in production, you can delete orders.
     * users: Orders can be deleted in production as well, if permission has been given.
     */
    'orders_deletable'  => env('BUTIK_ORDERS_DELETABLE', 'development'),

    /**
     * WIDGETS.
     *
     * Settings for your dashboard widgets
     */
    'widget_orders-limit' => 10,

    /**
     * ROUTING.
     *
     * Customize our custom routes as you need them. You can deactive some routes at all, if you want to
     * use your custom implementation or just rename our routes at all.
     */
    'shop_route_active'     => true, // default: true. Set to false to deactive this route.
    'product_route_active'  => true, // default: true. Set to false to deactive this route.

    'route_shop-prefix'       => 'shop',                       // yourshop.com/shop/
    'route_product'           => '{product}/{variant?}',       // yourshop.com/shop/xxxxx
    'route_cart'              => 'cart',                       // yourshop.com/shop/cart
    'route_checkout-delivery' => 'checkout/delivery',           // yourshop.com/shop/checkout/delivery
    'route_checkout-payment'  => 'checkout/payment',            // yourshop.com/shop/checkout/payment
    'route_payment-receipt'   => 'payment/{order}/receipt',     // yourshop.com/shop/payment/xxxxx/receipt

    /**
     * TEMPLATES.
     *
     * Define your own templates for the frontend if you want. You can as well edit the
     * given layouts to fit your needs.
     */
    'template_product-index'            => 'butik::web.shop.index',
    'template_product-show'             => 'butik::web.shop.show',
    'template_cart'                     => 'butik::web.cart.index',
    'template_checkout-delivery'        => 'butik::web.checkout.delivery',
    'template_checkout-payment'         => 'butik::web.checkout.payment',
    'template_checkout-receipt'         => 'butik::web.checkout.receipt',
    'template_checkout-receipt-invalid' => 'butik::web.checkout.invalid-receipt',

    /**
     * LAYOUTS.
     *
     * Define your own layouts for the frontend if you want. You can as well edit the
     * given layouts to fit your needs.
     */
    'layout_product-index'              => 'layout',
    'layout_product-show'               => 'layout',
    'layout_cart'                       => 'layout',
    'layout_checkout-delivery'          => 'layout',
    'layout_checkout-payment'           => 'layout',
    'layout_checkout-receipt'           => 'layout',
    'layout_checkout-receipt-invalid'   => 'layout',

    /**
     * SHIPPING.
     *
     * If you want, you can implement your own shipping methods or disable those, you don't want to use.
     * https://butik.dev/configuration/shipping
     */
    'shipping' => [
        'price' => \Jonassiewertsen\Butik\Shipping\ShippingByPrice::class,
        'per-item' => \Jonassiewertsen\Butik\Shipping\ShippingPerItem::class,
    ],

    /**
     * PAYMENT.
     *
     * If you want, you can swap our payment gateway with your own integration.
     * https://butik.dev/extending/payment-gateway
     */
    'payment_gateway' => Jonassiewertsen\Butik\Http\Controllers\PaymentGateways\MolliePaymentGateway::class,
];
